src/plugins/dashboard/public/application/dashboard_app_empty.tsx

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React from 'react';
+import {
+  AnalyticsNoDataPageKibanaProvider,
+  AnalyticsNoDataPage,
+} from '@kbn/shared-ux-page-analytics-no-data';
+import { DataPublicPluginStart } from '@kbn/data-plugin/public';
+
+import { DashboardAppServices } from '../types';
+import { useKibana } from '../services/kibana_react';
+
+export const DashboardAppEmpty = ({ onDataViewCreated }: { onDataViewCreated: () => void }) => {
+  const {
+    services: { core, data, dataViewEditor },
+  } = useKibana<DashboardAppServices>();
+  const analyticsServices = {
+    coreStart: core as unknown as React.ComponentProps<
+      typeof AnalyticsNoDataPageKibanaProvider
+    >['coreStart'],
+    dataViews: data.dataViews,
+    dataViewEditor,
+  };
+  return (
+    <AnalyticsNoDataPageKibanaProvider {...analyticsServices}>
+      <AnalyticsNoDataPage onDataViewCreated={onDataViewCreated} />;
+    </AnalyticsNoDataPageKibanaProvider>
+  );
+};
+
+export const isDashboardAppEmpty = async (dataViews: DataPublicPluginStart['dataViews']) => {
+  const hasUserDataView = await dataViews.hasData.hasUserDataView().catch(() => false);
+  const hasEsData = await dataViews.hasData.hasESData().catch(() => true);
+  if (!hasUserDataView || !hasEsData) {
+    return true;
+  }
+
+  const defaultDataView = await dataViews.getDefaultDataView();
```

Comment:
is this check necessary for Dashboard?

Reply:
That's a good question. The older version of this code had an `ensureDefaultDataView` check, so it seems to have been required before, but I can't exactly understand why. If there are _any_ data views, surely one of them should be default, no?

Is there a reason this was required for Discover?

Reply:
if memory serves well, in Discover, you save a saved search with a data view. if all data views get deleted and you try opening a saved search, that would throw an error. That's why the check. That being said, it was there before checking for existence of any data view, so I am not sure if it's relevant any more.

Reply:
Seems like it should all work the same. I will remove this from the dashboard side!